spec/runners/unified.rb

```diff
@@ -63,6 +63,7 @@ def define_unified_spec_tests(base_path, paths, expect_failure: false)
                 test.set_initial_data
                 lambda do
                   test.run
+                  skip test.skip_reason if test.skip?
```

Comment:
I don't know of a way to skip inside the test, so I set the `skip_reason` and return from the test then have the actual skip happen right after.

Reply:
My question is why skipping happens after `test.run`? Does it mean that we execute the test regardless of presence of a skip reason, and then just do not validate results?

Reply:
Well no, the idea is that the reason to skip happens inside the test. So in order to actually execute the skip I exit from the test early and call the skip outside of it (since skip is not accessible inside the test)

Reply:
Oh, got it, thank you for the explanation!